### PR TITLE
governance: initialize g.actualGovernanceBlock so nil cannot be set

### PR DESCRIPTION
**Pull request by yoomee1313**
## Proposed changes

- By https://github.com/klaytn/klaytn/pull/1117, `g.actualGovernanceBlock` can be nil because it is only set when `governanceBlock >= governanceStateBlock`
- So, this PR initializes `g.actualGovernanceBlock` first, and then check `governanceBlock >= governanceStateBlock`.
- Also, this PR restores the currentSet import from itemCache[governanceBlock] when `governanceBlock >= governanceStateBlock`. It seems unnecessary, but I restored it for cautious. It is from https://github.com/klaytn/klaytn/pull/927.

## Types of changes

Please put an x in the boxes related to your change.

- [x] Bugfix
- [ ] New feature or enhancement
- [ ] Others

## Checklist

*Put an x in the boxes that apply. You can also fill these out after creating the PR. If you're unsure about any of them, don't hesitate to ask. We're here to help! This is simply a reminder of what we are going to look for before merging your code.*

- [x] I have read the [CONTRIBUTING GUIDELINES](https://github.com/klaytn/klaytn/blob/master/CONTRIBUTING.md) doc
- [x] I have signed the [CLA](https://cla-assistant.io/klaytn/klaytn)
- [x] Lint and unit tests pass locally with my changes (`$ make test`)
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] I have added necessary documentation (if appropriate)
- [ ] Any dependent changes have been merged and published in downstream modules

## Related issues

- https://github.com/klaytn/klaytn/issues/1106

## Further comments

If this is a relatively large or complex change, kick off the discussion by explaining why you chose the solution you did and what alternatives you considered, etc...

### CODE REVIEW

**Comment by yoomee1313** [comment]
@aeharvlee @jeongkyun-oh please take a look.

File: governance/default.go [hunk 1]
```diff
@@ -667,18 +667,21 @@ func (g *Governance) initializeCache() error {
...
-		g.actualGovernanceBlock.Store(newBlockNumber)
+	}
+	newBlockNumber, newGovernanceSet, _ := g.ReadGovernance(headBlockNumber)
+	g.actualGovernanceBlock.Store(newBlockNumber)
+	g.currentSet.Import(newGovernanceSet)
+
+	governanceBlock, governanceStateBlock := g.actualGovernanceBlock.Load().(uint64), atomic.LoadUint64(&g.lastGovernanceStateBlock)
+	if governanceBlock >= governanceStateBlock {
+		ret, _ := g.itemCache.Get(getGovernanceCacheKey(governanceBlock))
+		g.currentSet.Import(ret.(map[string]interface{}))
```
Comment (aidan-kwon) [inline governance/default.go:684]: Is this code safe when `ret` is `nil` ?

Comment (yoomee1313) [reply governance/default.go:688]: Yes, It's not safe. I added error handling logic at https://github.com/klaytn/klaytn/pull/1168/commits/6a50c4107d7b5929eeb518f9728aa7c63aa12de4. Please take a look.

File: governance/default.go [hunk 2]
```diff
@@ -667,18 +667,25 @@ func (g *Governance) initializeCache() error {
...
+	// head block number is used to get the appropriate g.currentSet and g.actualGovernanceBlock
+	headBlockNumber := uint64(0)
+	if headBlockHash := g.db.ReadHeadBlockHash(); !common.EmptyHash(headBlockHash) {
+		if num := g.db.ReadHeaderNumber(headBlockHash); num != nil {
+			headBlockNumber = *num
+		}
+	}
+	newBlockNumber, newGovernanceSet, _ := g.ReadGovernance(headBlockNumber)
+	g.actualGovernanceBlock.Store(newBlockNumber)
+	g.currentSet.Import(newGovernanceSet)
```
Comment (aidan-kwon) [inline governance/default.go:679]: This code change `currentSet`, but doesn't call `g.updateGovernanceParams()`.

Comment (yoomee1313) [reply governance/default.go:679]: thanks, reflected at https://github.com/klaytn/klaytn/pull/1168/commits/b59117a9563072e6f6772d2cbebb74bd800efc13.

**Review by aidan-kwon — DISMISSED** [review]

File: governance/default.go [hunk 3]
```diff
@@ -667,18 +667,26 @@ func (g *Governance) initializeCache() error {
...
-				headBlockNumber = *num
-			}
+	// head block number is used to get the appropriate g.currentSet and g.actualGovernanceBlock
+	headBlockNumber := uint64(0)
+	if headBlockHash := g.db.ReadHeadBlockHash(); !common.EmptyHash(headBlockHash) {
+		if num := g.db.ReadHeaderNumber(headBlockHash); num != nil {
+			headBlockNumber = *num
+		}
+	}
+	newBlockNumber, newGovernanceSet, _ := g.ReadGovernance(headBlockNumber)
```
Comment (jeongkyun-oh) [inline governance/default.go:677]: @yoomee1313 How about adding a comment about the point that we were confused? (e.g. difference from `g.db.ReadGovernance`, or why we put this out of the above for loop).

Don't we need to handle the returned error?

**Review by jeongkyun-oh — DISMISSED** [review]

**Comment by aidan-kwon** [comment]
@mckim19 Can you check this PR?

File: governance/default.go (hunk 3 above)
Comment (yoomee1313) [reply governance/default.go]: @jeongkyun-oh Thanks, I added error handling logic and some comments to clarify the function's behavior. Please take a look! https://github.com/klaytn/klaytn/pull/1168/commits/55ad79072d91478eaae2f86e46664ebfa6c4b0d0

**Review by aidan-kwon — APPROVED** [review]

**Review by jeongkyun-oh — APPROVED** [review]

**Review by hqjang-pepper — APPROVED** [review]

**Pull request merged**